src/lib.rs

```diff
@@ -732,6 +732,29 @@ pub trait Itertools : Iterator {
         adaptors::unique(self)
     }
 
+    /// Check whether all elements compare equal.
+    ///
+    /// Empty iterators are considered to have equal elements:
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![1, 1, 1, 2, 2, 3, 3, 3, 4, 5, 5];
+    /// assert!(!data.iter().all_equal());
+    /// assert!(data[0..3].iter().all_equal());
+    /// assert!(data[3..5].iter().all_equal());
+    /// assert!(data[5..8].iter().all_equal());
+    ///
+    /// let data : Option<usize> = None;
+    /// assert!(data.into_iter().all_equal());
+    /// ```
+    fn all_equal(self) -> bool 
```

Comment:
Ok, it's not always easy to decide, but I think this method should use `&mut self` and no `Self: Sized`, just like `all`. What do you think?